src/formatFiles.js

```diff
@@ -1,27 +1,51 @@
 import { readFileSync, writeFileSync } from 'fs';
-import { resolveConfig, format } from 'prettier';
+import { resolveConfig, format as prettierFormat } from 'prettier';
 import { join } from 'path';
 
 export default (
-  directory,
+  rootDirectory,
   files,
   { config, onWriteFile, onExamineFile } = {}
 ) => {
+  // Use local "prettier-tslint" if possible.
+  const tslintFormat = loadFormat(rootDirectory, 'prettier-tslint', 'format');
+  // Use local "prettier-eslint" if possible.
+  const eslintFormat = loadFormat(rootDirectory, 'prettier-eslint');
+
   for (const relative of files) {
     onExamineFile && onExamineFile(relative);
-    const file = join(directory, relative);
-    const options = resolveConfig.sync(file, { config, editorconfig: true });
+    const file = join(rootDirectory, relative);
     const input = readFileSync(file, 'utf8');
-    const output = format(
-      input,
-      Object.assign({}, options, {
-        filepath: file,
-      })
-    );
+
+    const format =
+      (/\.(mjs|jsx?)$/.test(file) && (eslintFormat || tslintFormat)) ||
+      (/\.tsx?$/.test(file) && (tslintFormat || eslintFormat)) ||
+      prettierFormat;
+
+    const options = resolveConfig.sync(file, { config, editorconfig: true });
+    const output =
+      format === prettierFormat
+        ? format(input, ((options.filepath = file), options))
+        : format({
+            text: input,
+            filePath: file,
+            prettierOptions: options,
+          });
 
     if (output !== input) {
       writeFileSync(file, output);
       onWriteFile && onWriteFile(relative);
     }
   }
 };
+
+function loadFormat(rootDirectory, formatName, exportName) {
+  let entry;
+  try {
+    entry = require.resolve(join(rootDirectory, 'node_modules', formatName));
```

Comment:
This won't work with yarn PnP.

Reply:
Are you saying it breaks `pretty-quick` for Yarn PnP installs, or just that `prettier-eslint` and `prettier-tslint` won't be found?

Reply:
Have you actually tested this with Yarn PnP?

Reply:
The latter. Projects using PnP don't have `node_modules`

Reply:
Okay, that's fine. We can patch PnP support for this feature in a later patch.

Reply:
I've pushed a commit that resolves this.